1_data/src/subset_nwm.R

```diff
@@ -99,7 +99,27 @@ subset_nwm <- function(ind_file, model_configuration, comids, gd_config) {
     }
   }
 
-  ncvar_put(new_nc, new_nc$var$streamflow, streamflow)
+  # need to loop through the sites because the dimensions in streamflow matrix were written incorrectly
+  for(s in 1:length(site_inds)) {
+    if(length(dimids) == 2) {
+      # Note axis order is assumed here!!!
+      ncvar_put(new_nc,
+                new_nc$var$streamflow,
+                streamflow[,s]*(1/new_nc$var$streamflow$scaleFact),  # need to multiply streamflow data by scale factor in nc var
```

Comment:
Ugh... does ncdf4 apply the scale_factor on read but not on write?

Reply:
It seems that way...not at all intuitive, and it'd be great if you could test that with the nc data straight from THREDDS, but with our made-up data this seemed to be necessary in just one direction. The help file for ncvar_get says,
> If the variable in the netCDF file has a scale and/or offset attribute defined, the returned data are automatically and silently scaled and/or offset as requested.

while the help file for ncvar_put doesn't mention "scale" at all. https://cran.r-project.org/web/packages/ncdf4/ncdf4.pdf

Looks like another option might be to use raw_datavals=TRUE` in the call to ncvar_get...but I just noticed that, and I think the rescaling option we implemented above does work. I hope.

Reply:
It seems like it's applied on write but not read?? what we put in was scaled by the factor and before closing nc, reading with ncvar_get() showed a scaled version of what we ncvar_put()